### PR TITLE
MultiRegion append_regions preserves provenance

### PR DESCRIPTION
**Pull request by TomGoBravo**
* change `MultiRegionTimeseriesDataset.provenance` to never be None and always sorted, simplifying code that uses it
* in `MultiRegionTimeseriesDataset.append_regions` also append the provenance data.

This PR addresses https://trello.com/c/7tgdgPUV/547-provenance-file-stopped-updating-oct-28

### CODE REVIEW

**Comment by TomGoBravo** [comment]
[Provenance file stopped updating Oct 28](https://trello.com/c/7tgdgPUV/547-provenance-file-stopped-updating-oct-28)

File: libs/datasets/timeseries.py [hunk 1]
```diff
@@ -567,18 +583,27 @@ def __post_init__(self):
...
-            pd.concat([self.data, other.data], ignore_index=True)
-        ).append_latest_df(
-            pd.concat(
-                [self.latest_data.reset_index(), other.latest_data.reset_index()], ignore_index=True
+        return (
+            MultiRegionTimeseriesDataset.from_timeseries_df(
+                pd.concat([self.data, other.data], ignore_index=True)
+            )
+            .append_latest_df(
+                pd.concat(
```
Comment (ghop02) [inline libs/datasets/timeseries.py:597]: definitely not necessary, but making these dataframes their own variables would make this block more concise

**Review by ghop02 — APPROVED** [review]
Seems good to me! would prefer updating the files in data/ via the github action, but not too worried

File: libs/datasets/timeseries.py (hunk 1 above)
Comment (TomGoBravo) [reply libs/datasets/timeseries.py]: Done in a0bdee18

**Pull request merged**